jgrapht-core/src/main/java/org/jgrapht/alg/TransitiveReduction.java

```diff
@@ -131,11 +137,20 @@ static void transitiveReduction(BitSet[] pathMatrix)
      * @param directedGraph the directed graph that will be reduced transitively
      * @param <V> the graph vertex type
      * @param <E> the graph edge type
+     * @throws IllegalArgumentException if the given directed graph contains cycles because this algorithm only works
+     *                                  for acyclical graphs
      */
     public <V, E> void reduce(final Graph<V, E> directedGraph)
     {
         GraphTests.requireDirected(directedGraph, "Graph must be directed");
 
+        if (new CycleDetector<>(directedGraph).detectCycles()) {
```

Comment:
Okay, I think I can make the check optional by overloading `reduce` by a method with an additional boolean parameter.

What I wanted to discuss, though, is that someone mentioned before he did not like the singleton pattern in this class, which is why I implemented it differently in the other class. This is inconsistent. Should this be unified? But it would result in a breaking change here. But actually, improving existing classes was not my point here. The reason I did add the check and also the additional unit tests here is that I wanted to make this class aware of its own limits and also tell the user about it.